colexec: support TIMESTAMPTZ type

yuzefovich opened the pull request:
**colexec: support TIMESTAMPTZ type**

This commit adds the support for TimestampTZ data type which is
represented in the same way as Timestamp (as 'time.Time'). We already
had everything in place, so only the type-conversion was needed.

Addresses: #42043.

Release note (sql change): vectorized engine now supports TIMESTAMPTZ
data type.

**sqlsmith: add several types to vecSeedTable**

This commit adds previously supported INT2 and INT4 types to
vecSeedTable as well as newly supported TIMESTAMPTZ.

Release note: None

yuzefovich committed:
colexec: support TIMESTAMPTZ type

This commit adds the support for TimestampTZ data type which is
represented in the same way as Timestamp (as 'time.Time'). We already
had everything in place, so only the type-conversion was needed.

Release note (sql change): vectorized engine now supports TIMESTAMPTZ
data type.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43514)
<!-- Reviewable:end -->

yuzefovich committed:
sqlsmith: add several types to vecSeedTable

This commit adds previously supported INT2 and INT4 types to
vecSeedTable as well as newly supported TIMESTAMPTZ.

Release note: None

jordanlewis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/43514#-:-Lx2dt1nANXhfSNjBZGx:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
